packages/x-date-pickers/src/internals/components/PickersPopper.tsx

```diff
@@ -280,13 +280,22 @@ export const PickersPopper = (props: PickerPopperProps) => {
     ...otherPaperProps
   } = PaperProps;
 
+  const handleKeyDown = (event) => {
+    if (event.key === 'Escape') {
```

Comment:
From what I understand, this is used because it's adding an event with `document.addEventListener(` then we have only access to a native event.

With `onKeyDown` we get a Synthetic event, so React is responsible for having consistent properties across browsers